### ISSUE TITLE
Filter Form Accessibility Issues

### ISSUE THREAD

**Issue by slowbot** [CityOfDetroit/bloom#1243]
- [x] Image card on browse alt text should be set to either blank “” or made specific to each listing https://detroit-public-prod.netlify.app/listings
- [x] Filter form label aria describedby error attribute should not display until the field is in error. Displaying the attribute without the error message is invalid https://detroit-public-prod.netlify.app/listings
- [x] Closed button and Rent range fields should use screen reader only labels rather than have an empty label https://detroit-public-prod.netlify.app/listings
